Snakemake/workflow/rules/prepare_files_for_tsinfer.smk

```diff
@@ -1,115 +1,116 @@
+if config['ancestral_allele'] is not None:
+    ancestral_file = config['ancestral_allele']
+else:
+    ancestral_file = rules.combine_pos_ancestral.output	
+
+
+def getInputVcfFile_af(wildcards):
+    if os.path.isfile(vcfdir + "/" + wildcards.chromosome + '_phased.vcf.gz'):
+        print("Phased file found")
+        vcf_file = vcfdir + "/" + wildcards.chromosome + '_phased.vcf.gz'
+        print(vcf_file)
+    else:
+        print("Phased file not found")
+        file = open(vcfdir + '/Vcf_file_' + wildcards.chromosome + '.txt')
+        vcf_file = file.read().strip("\n")
+        print(vcf_file)
+    return(vcf_file)
+
 rule get_af:
     input: f'{vcfdir}/{{chromosome}}_phased.vcf.gz'
     output:
-        new_file = temp(f'{vcfdir}/{{chromosome}}_phased_info.vcf.gz'), 
         info = temp(f'{vcfdir}/Info{{chromosome}}.INFO')
     params:
         prefix=f'{vcfdir}/Info{{chromosome}}'
     conda: "bcftools"
     threads: 1
-    resources: cpus=1, mem_mb=4000, time_min=5
+    resources: cpus=1, mem_mb=32000, time_min=60
     log: 'logs/get_af_{chromosome}.log'
     shell:
         """
-        bcftools +fill-tags {input} -Oz -o {output.new_file} -- -t AN,AC,AF
-        vcftools --gzvcf {output.new_file} --out {params.prefix} --get-INFO AC --get-INFO AF
-	
-	LOGFILE={params.prefix}.log
-	if test -f "$LOGFILE"; then
-	    rm $LOGFILE
-	fi
+        bcftools +fill-tags {input} -Oz -o {input} -- -t AN,AC,AF
+        vcftools --gzvcf {input} --out {params.prefix} --get-INFO AC --get-INFO AF
+
+        LOGFILE={params.prefix}.log
+        if test -f "$LOGFILE"; then
+            rm $LOGFILE
+        fi
         """
 
 rule get_major:
     input: rules.get_af.output.info
-    output: temp(f'{vcfdir}/Major{{chromosome}}.txt')
+    output: temp(f'{vcfdir}/Major{{chromosome}}.txt')  
     threads: 1
-    resources: cpus=1, mem_mb=4000, time_min=5
-    log: 'logs/get_major_{chromosome}.log'
+    resources: cpus=1, mem_mb=32000, time_min=60
+    log: 'logs/Get_major_{chromosome}.log'
+#    wildcard_constraints:
+#        chromosome="^chr | ^Chr"
     shell:
         """
         awk '{{if (NR!=1 && $5>=0.5) {{print $1"_"$2","$4}} else if (NR!=1 && $5<0.5) {{print $1"_"$2","$3}}}}' {input} > {output}
         """
 
-rule decompress:
+rule extract_ancestral_chromosome:
     input:
```

Comment:
This was added to extract the chromosomal ancestral file from the full ancestral file